doc/guides/west/manifest.rst

```diff
@@ -1618,10 +1618,16 @@ Here, ``west update`` will initialize and update all submodules in ``foo``. If
 Option 2: List of mappings
 ==========================
 
-The ``submodules`` key may be a list of mappings. Each element in the list
-defines the name of the submodule to update, and the path -- relative to the
-project's absolute path in the workspace -- to use for the submodule.
-The submodule will be updated recursively.
+The ``submodules`` key may be a list of mappings, one list element for
+each desired submodule. Each submodule listed is updated recursively.
+You can still track and update unlisted submodules with ``git`` commands
+manually; present or not they will be completely ignored by ``west``.
+
+The ``path`` key must match exactly the path of one submodule relative
+to its parent west project, as shown in the output of ``git submodule
+status``. The ``name`` key is optional and not used by west for now;
+it's not passed to ``git submodule`` commands either. The ``name`` key
+was briefly mandatory in west version 0.9.0, but was made optional in 0.9.1.
```

Comment:
Not sure this is the right place to comment, but doesn't this require a manifest version change, due to API breaking ?
And hence a `west` 0.10.0, and not just 0.9.1.

Imagine I personally use `west` 0.9.1 and create a manifest `west.yml` WITHOUT the name key.
Everything works fine, but userA with `west` version 0.9.0 now tries to fetch my manifest repo, but because `west` 0.9.0 has `name` key as mandatory, then the schema validation / `west update` will fail.

I consider that a schema breaking change.

Reply:
Good catch, thanks. https://docs.zephyrproject.org/latest/guides/west/manifest.html#version
> The version subsection can be used to mark the lowest version of the manifest file schema that can parse this file’s data

Making a key optional is not that much different from adding a new key.

Would a 0.9.1 manifest version also provide 0.9.0 users the expect error message? That documentation does not attach any semantic to digits and does not show any example with 3 digits either.

The other, much simpler option is to release a 0.9.1 "hotfix" and tell everyone to stop using 0.9.0 that was released a very short time ago. This would help a lot with https://github.com/zephyrproject-rtos/west/pull/481/ too.

> Not sure this is the right place to comment

Maybe https://github.com/zephyrproject-rtos/west/pull/479 would be a better place. I think it's past time to move west documentation to the west repo.

Reply:
> Would a 0.9.1 manifest version also provide 0.9.0 users the expect error message? That documentation does not attach any semantic to digits and does not show any example with 3 digits either.

I believe the original intentions by @mbolivar-nordic was to only use `x.y` for the schema version, and use third digit only in `west` releases that didn't change the manifest schema in api breaking ways.



> > Not sure this is the right place to comment
> 
> Maybe [zephyrproject-rtos/west#479](https://github.com/zephyrproject-rtos/west/pull/479) would be a better place.

except for the fact that [zephyrproject-rtos/west#479](https://github.com/zephyrproject-rtos/west/pull/479) is already merged. I want to ensure that if a new `west` release is made, then we jump to the right version, 0.9.1 vs 0.10.0.
Placing a comment in a merged PR doesn't get the right attention.

Reply:
> The other, much simpler option is to release a 0.9.1 "hotfix" and tell everyone to stop using 0.9.0 that was released a very short time ago. This would help a lot with [zephyrproject-rtos/west#481](https://github.com/zephyrproject-rtos/west/pull/481) too.

Problem here is that there are already releases and SDKs being made based on west 0.9.0.

Reply:
> I want to ensure that if a new west release is made, then we jump to the right version, 0.9.1 vs 0.10.0.
> Placing a comment in a merged PR doesn't get the right attention.

If a sub-thread about west in a zephyr doc PR is more likely to make sure then why not. Maybe @mbolivar-nordic would prefer a west issue?

> Problem here is that there are already releases and SDKs being made based on west 0.9.0.

0.9.0 was tagged less than 3 weeks ago. For the group filter issue 481 it's honestly not a good idea to make releases and SDKs rely on some totally new and major feature of any dependency without testing that feature much before the release - not unless you can quickly release updates yourself. I mean you can't have a "slow" dependant depend on "fast" dependencies, that's bound to fail exactly like this.

Back to this optional `name` topic, neither adding that key in the manifest nor upgrading from 0.9.0 to 0.9.1 seems like a big deal. So yes, this is technically an incompatible change but will anyone really care in practice? Or have we already spent more time discussing this than anyone else will ever? (the group-filter issue is more complicated)

Reply:
> For the group filter issue 481 it's honestly not a good idea to make releases and SDKs rely on some totally new and major feature of any dependency without testing that feature much before the release - not unless you can quickly release updates yourself.

Well, for a start we are no longer relying on the group filter issue as we removed that dependency, but that doesn't mean there are not other west 0.9.0 features that can be used. And it was exactly because we were testing this before our release (even before rc1) that we discovered the issue, and could address it accordingly.
Packaging a set of tools for easy deployment doesn't seem wrong to me, but that set of tools of course can only include what is available.

I think the `group-filter` could be seen as a hotfix, but breaking the schema version would be a new release imho.
But let's hear @mbolivar-nordic opinion.

> Maybe @mbolivar-nordic would prefer a west issue?

hard to make an issue on something that is not an issue.
If @mbolivar-nordic was making a `west` 0.9.1 release PR, then I could make a comment there, but if he cuts a `0.10.0` directly, then there is no issue.

Reply:
Hi, thanks for the ping. I agree with @tejlmand that we can't put this into 0.9 for the reasons outlined (it really ought to be a schema bump).

Let's get this into 0.10. The PRs for that are already up:

https://github.com/zephyrproject-rtos/zephyr/pull/32381
https://github.com/zephyrproject-rtos/west/pull/482

I'm preparing some backports to 0.9 for some fixes in 482 and I will revert the submodule change in there. Sorry @marc-hb , this one is my bad; I should have thought of this.